source/common/router/router.h

```diff
@@ -192,6 +185,21 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     void onPoolReady(Http::StreamEncoder& request_encoder,
                      Upstream::HostDescriptionConstSharedPtr host) override;
 
+    void setRequestEncoder(Http::StreamEncoder& request_encoder);
+    void clearRequestEncoder();
+
+    struct DownstreamWatermarkManager : public Http::DownstreamWatermarkCallbacks {
```

Comment:
The Http::StreamCallbacks and the DownstreamWatermarkCallbacks have the same names.  I could rename one of them?